### PR TITLE
Improve README organization and content update

### PR DESCRIPTION
**Pull request by jasonburkert-ccpo**
Major re-organization under these headers:

- Install Prerequisites
- Deploy Shared Infrastructure
- Deploy Azure Function Apps
- Support Information

Made _Deploy..._ sections generic to include `./poc` and newer `./provisioning`

### CODE REVIEW

**Comment by jasonburkert-ccpo** [comment]
Not sure how table under `Install Prerequisites` will be received, due to reduced readability in text-only viewing.  I think it helps when rendered.

Happy with the rest of it.

File: README.md [hunk 1]
```diff
@@ -4,165 +4,172 @@ The ATAT Web API is the internal application programming interface that supports
...
+| Terraform | n/a | Follow steps at [Install Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/azure-get-started) at hashicorp.com |
+
+### Deploy Shared Infrastructure
 ---
+[Terraform](https://www.terraform.io/) will be used to deploy shared infrastructure in Microsoft Azure.
+Examples of such shared resources:
+* Key Vault
+* Storage Account
+* Cosmos DB Account and Database 
+* Enterprise Service Bus
```
Comment (jeffsegal-ccpo) [inline README.md:27]: s/Enterprise/Azure

File: README.md [hunk 2]
```diff
@@ -4,165 +4,172 @@ The ATAT Web API is the internal application programming interface that supports
...
-
-### Install dependencies
-For a detailed tutorial, follow the guidance here: [Install Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/azure-get-started)
-
-### Local Deployment
-
-Navigate to the `./poc/shared_infrastructure` directory.
+The following is a list of `main.tf` files and their locations which can be used with Terraform to deploy Shared Infrastructure:
+* `./provisioning/shared_infrastructure/main.tf`
+* `./poc/shared_infrastructure/main.tf`
```
Comment (jeffsegal-ccpo) [inline README.md:31]: Now that we have the real API, let's remove the documentation for the PoC (we can keep the actual files around for a bit longer though).

File: README.md [hunk 3]
```diff
@@ -4,165 +4,172 @@ The ATAT Web API is the internal application programming interface that supports
...
+| ----------- | ----------- |----------- |
+| Node Version Manager | `curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.38.0/install.sh \| bash` | Then restart your shell. Details available in the [nvm README](https://github.com/nvm-sh/nvm#installing-and-updating). |
+| Node.js | `nvm install` | Node.js version is specified in file `./.nvmrc` |
+| Serverless Framework | `npm install -g serverless` | n/a |
+| TypeScript | `npm install -g typescript` | n/a |
+| Azure CLI | n/a | Follow steps at [Install the Azure CLI](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli) at microsoft.com |
+| Terraform | n/a | Follow steps at [Install Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/azure-get-started) at hashicorp.com |
+
+### Deploy Shared Infrastructure
 ---
```
Comment (jeffsegal-ccpo) [inline README.md:21]:
```suggestion
---
This monorepo currently contains a single Application (ATAT Portfolio Drafts API), but may eventually contain multiple Applications which all share some common infrastructure pieces. The following instructions describe their deployment:
```

File: README.md [hunk 4]
```diff
@@ -4,165 +4,172 @@ The ATAT Web API is the internal application programming interface that supports
...
 
-#### Create Azure Service Principal and set environment variables
+### Deploy Azure Function Apps
+---
+[Serverless Framework](https://www.serverless.com/) will be used to deploy Azure Function Apps.
 
-Follow steps at [Creating a Service Principal](https://www.serverless.com/framework/docs/providers/azure/guide/quick-start#creating-a-service-principal) at serverless.com
+The following is a list of `serverless.yml` files and their locations which can be used with Serverless Framework to deploy Function Apps:
+* `./provisioning/serverless.yml`
+* `./poc/random_quote/serverless.yml`
```
Comment (jeffsegal-ccpo) [inline README.md:87]: Ditto to above comment, can remove PoC items.

File: README.md [hunk 5]
```diff
@@ -4,165 +4,172 @@ The ATAT Web API is the internal application programming interface that supports
...
 
-To destroy the deployed shared_infrastructure run:
+To destroy the deployed shared infrastructure run:
 
 ```
 terraform destroy
 ```
 
-#### Create Azure Service Principal and set environment variables
+### Deploy Azure Function Apps
```
Comment (jeffsegal-ccpo) [inline README.md:81]:
```suggestion
### Deploy ATAT Portolio Drafts API with Serverless Framework
```

**Review by jeffsegal-ccpo — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (jasonburkert-ccpo) [reply README.md:27]: Thank you.  Fixed.

File: README.md (hunk 2 above)
Comment (jasonburkert-ccpo) [reply README.md:31]: Thank you.  Removed doc references.

File: README.md (hunk 4 above)
Comment (jasonburkert-ccpo) [reply README.md:87]: Affirmative.

File: README.md [hunk 6]
```diff
@@ -4,165 +4,171 @@ The ATAT Web API is the internal application programming interface that supports
...
-#### Retrieve .tfstate from Azure
+##### Select a workspace
 
-Select the correct workspace inside the container on Azure
+Select the desired workspace.
+Terraform state is stored by default in a file named _terraform.tfstate_.  In this case it is stored remotely inside the storage container on Azure.
 
 ```
-terraform workspace select <name of workspace>
+terraform workspace select <your-workspace-name>
```
Comment (anhtran-ccpo) [inline README.md:54]: Should we add a `terraform new <your-workspace-name>` step before this step if they haven't made one yet?

Comment (jasonburkert-ccpo) [reply README.md:54]: I like that idea.

File: README.md [hunk 7]
```diff
@@ -4,165 +4,171 @@ The ATAT Web API is the internal application programming interface that supports
...
-## Shared Infrastructure
-Shared infrastructure for the PoC can be found at `./poc/shared_infrastructure`
-
-### Install dependencies
-For a detailed tutorial, follow the guidance here: [Install Terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli?in=terraform/azure-get-started)
-
-### Local Deployment
-
-Navigate to the `./poc/shared_infrastructure` directory.
+The following is a list of `main.tf` files and their locations which can be used with Terraform to deploy Shared Infrastructure:
```
Comment (jeffsegal-ccpo) [inline README.md:30]: We can tidy this up to just point to the single `main.tf` now

**Review by anhtran-ccpo — APPROVED** [review]

**Review by zachclark-ccpo — APPROVED** [review]

File: README.md (hunk 6 above)
Comment (jasonburkert-ccpo) [reply README.md:59]: Added.

File: README.md (hunk 7 above)
Comment (jasonburkert-ccpo) [reply README.md]: Did this for navigation instructions in both areas where `./poc` references were removed.

**Comment by jasonburkert-ccpo** [comment]
All requested changes have been incorporated.  This is ready for merging or the next round of review.

**Review by jeffsegal-ccpo — APPROVED** [review]

**Pull request merged**